src/network/networkd-link.c

```diff
@@ -2567,6 +2575,9 @@ static int link_carrier_gained(Link *link) {
 
         assert(link);
 
+        if (link->iftype == ARPHRD_CAN)
+                return link_handle_bound_by_list(link);
```

Comment:
maybe add comment here, `/* let's shortcut things for CAN which doesn't need most of what#s done below. */` or something like that?